internal/controllers/core/kubernetesdiscovery/portforwards.go

```diff
@@ -23,7 +23,7 @@ func (r *Reconciler) manageOwnedPortForwards(ctx context.Context, nn types.Names
 	err := indexer.ListOwnedBy(ctx, r.ctrlClient, &pfList, nn, apiType)
 	if err != nil {
 		return fmt.Errorf("failed to fetch managed PortForward objects for KubernetesDiscovery %s: %v",
-			kd.Name, err)
+			nn.Name, err)
```

Comment:
This caused a panic while testing. There's already a comment a few lines up saying `kd` might be nil during deletion.